### ISSUE TITLE
Unicode characters are not rendering when exported to Android

### ISSUE THREAD

**Issue by phil-hudson** [godotengine/godot#77903]
### Godot version

4.0.3

### System information

Windows 11, Android Samsung A14

### Issue description

When rendering a label with Chinese characters, on the exported Android app, they render as the hex value instead of the actual character.

I am using a font that supports these characters (noto-sans), and they render correctly when being run locally on PC, as well as exported to iOS.

### Steps to reproduce

Run the reproduction project. Export to android, you will see the error case. Export / run locally or to iOS and you will see that it works correctly.

Please see screenshots attached of the reproduction project:

Error case from android device:

![Screenshot_20230606_174126_unicode_android_debug](https://github.com/godotengine/godot/assets/1914950/f569e461-f9ed-4627-a84b-cb31c66dffed)

Local run on PC:

![image](https://github.com/godotengine/godot/assets/1914950/afca294f-f0ce-43d8-9cba-5649864a581e)

### Minimal reproduction project

[unicode_android_debug.zip](https://github.com/godotengine/godot/files/11663011/unicode_android_debug.zip)

**Comment by phil-hudson** [comment]
> In the future make sure to follow the instructions and exclude the `.godot` folder, it helps with size and security

updated

**Comment by bruvzg** [comment]
> I am using a font that supports these characters (noto-sans), and they render correctly when being run locally on PC, as well as exported to iOS.

No, your project does not have CJK font, only base Latin/Greek/Cyrillic Noto-Sans file, so system fonts are used as fallback and which characters are rendered depends on available system fonts.

**Comment by bruvzg** [comment]
Unusually, Android includes a full set of Noto fonts, and fallback should work. So there might be issues with the fallback font detection (maybe non-standard system font path on Samsung devices?).

**Comment by phil-hudson** [comment]
> > I am using a font that supports these characters (noto-sans), and they render correctly when being run locally on PC, as well as exported to iOS.
> 
> No, your project does not have CJK font, only base Latin/Greek/Cyrillic Noto-Sans font file (full Noto-Sans have multiple files and almost 1.3 GB large), so system fonts are used as fallback and which characters are rendered depends on available system fonts.

Thanks so much for flagging, apologies for the issue. I downloaded the Noto font from the Noto SC page, but you're right it didn't include the full charset. Thank you again

**Comment by Calinou** [comment]
Looking at the above description, this may actually be a bug. I'll test this on a OnePlus 6 (which is close to stock Android) and Samsung Galaxy Z Fold4 to make sure.

**Comment by Calinou** [comment]
> I'm observing the same problem when I export the project to HTML5.

The Web platform doesn't support loading system fonts, so this is expected.
